### PR TITLE
refactor(Scripts/Onyxia): Update Onyxia's script

### PR DESCRIPTION
**Pull request by Nyeriah**
## Changes Proposed:
-  Moved her to BossAI
-  Updated instance to use BossStates
- Removed useless creature scripts

## Issues Addressed:

- Closes 

## Tests Performed:

- Tested in-game
- 

## How to Test the Changes:

1.Engage Onyxia
2. Check that her abilities & phase transitions still work properly
3. Check that achievements still work properly

## Known Issues and TODO List:

- [ ]
- [ ]

## How to Test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here on the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

**REMEMBER**: when testing a PR that changes something **generic** (i.e. a part of code that handles more than one specific thing), the tester should not only check that the PR does its job (e.g. fixing spell XXX) but **especially** check that the PR does not cause any regression (i.e. introducing new bugs).

**For example**: if a PR fixes spell X by changing a part of code that handles spells X, Y, and Z, we should not only test X, but **we should test Y and Z as well**.

### CODE REVIEW

**Comment by Kitzunu** [comment]
as you are refactoring it, please also update to new script system

**Comment by Nyeriah** [comment]
This is ready to be tested and reviewed

File: src/server/scripts/Kalimdor/OnyxiasLair/onyxias_lair.h [hunk 1]
```diff
@@ -58,4 +58,6 @@ inline AI* GetOnyxiasLairAI(T* obj)
     return GetInstanceAI<AI>(obj, OnyxiasLairScriptName);
 }
 
+#define RegisterOnyxiaLairCreatureAI(ai_name) RegisterCreatureAIWithFactory(ai_name, GetOnyxiasLairAI)
```
Comment (Kitzunu) [inline src/server/scripts/Kalimdor/OnyxiasLair/onyxias_lair.h:61]:
```suggestion
#define RegisterOnyxiasLairCreatureAI(ai_name) RegisterCreatureAIWithFactory(ai_name, GetOnyxiasLairAI)
```

**Review by Kitzunu — CHANGES_REQUESTED** [review]

**Comment by temperrr** [comment]
Tested (havent tested without refactor)

- [ ] Onyxia isn't tauntable
- [x] used flamebreath
- [x] used wing buffet
- [x] used tail sweep
- [x] used cleave
- [x] phase 2 correctly engaged
- [x] fireball
- [x] spawning guards and whelps
- [x] fire breath
- [x] phase 3 engaged
- [x] Fear
- [x] eruption
- [x] few whelps spawned

achievements
10 player:
- [x] Onyxias lair 10 player
- [x] Defeat Onyxia in less than 5 minutes
- [x] Cause 50 Onyxia Whelplings to hatch within 10 seconds of Onyxia's liftoff, and then defeat her
- [x] Defeat Onyxia without anyone taking damage from a Deep Breath

25 player

- [x] Onyxias lair 10 player
- [x] Defeat Onyxia in less than 5 minutes
- [x] Cause 50 Onyxia Whelplings to hatch within 10 seconds of Onyxia's liftoff, and then defeat her
- [x] Defeat Onyxia without anyone taking damage from a Deep Breath

**Review by Nefertumm — APPROVED** [review]

**Comment by Nefertumm** [comment]
Looking good for me. I'm just wondering why did you use a variable for phases instead of events.IsInPhase(...) but it's the same :) Good work

**Comment by Nyeriah** [comment]
> Looking good for me. I'm just wondering why did you use a variable for phases instead of events.IsInPhase(...) but it's the same :) Good work

I didn't use it, it's old code, since this is a side-project to me I didn't want to invest too much time rewriting that logic, it's working fine as it is

**Pull request merged**